gnat2goto/driver/tree_walk.adb

```diff
@@ -3862,16 +3862,19 @@ package body Tree_Walk is
    function Process_Statements (L : List_Id) return Irep is
       Reps : constant Irep := New_Irep (I_Code_Block);
       Stmt : Node_Id := First (L);
-
+      package IO renames Ada.Text_IO;
    begin
       while Present (Stmt) loop
          begin
             Process_Statement (Stmt, Reps);
          exception
             when Error : others =>
-                  Ada.Text_IO.Put_Line (Ada.Text_IO.Standard_Error,
-                                        Ada.Exceptions.Exception_Information
-                                          (Error));
+               IO.Put_Line (IO.Standard_Error, "<========================>");
+               IO.Put_Line (IO.Standard_Error,
+                            Ada.Exceptions.Exception_Information
+                              (Error));
+               Treepr.Print_Node_Subtree (Stmt);
```

Comment:
Is this likely to print information that a sensitive user might not want to send to us? If so, could we make this subtree dumping optional in some way?

Reply:
1. Yes
2. Yes

Reply:
At the moment I think we're mostly working on the assumption that the gnat frontend should be able to handle the likely Ada code that will be thrown at it, so we're mostly worried about knowing what the gant2goto bridge currently can't handle. @martin-cs probably has some thoughts too.

Reply:
@chrisr-diffblue Added `-no-dump-statement-ast-on-error` option to disable this

Reply:
@chrisr-diffblue I believe the GNAT front-end should be capable of handling all of the code involved.

@hannes-steffenhagen-diffblue This must be off by default.  I don't want omitting / forgetting / not realising a flag is needed to result in someone sending us something they shouldn't / sending more than we asked for.